src/EditorFeatures/CSharpTest/SplitOrMergeIfStatements/MergeConsecutiveIfStatementsTests.cs

```diff
@@ -0,0 +1,1648 @@
+        public async Task MergedOnElseIfCaret4()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if [||](b)
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfCaret5()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else if (b)[||]
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfCaret6()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        [||]else if (b)
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfCaret7()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else[||] if (b)
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfIfKeywordSelection()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else [|if|] (b)
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfHeaderSelection1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        else [|if (b)|]
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfHeaderSelection2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+        [|else if (b)|]
+        {
+        }
+    }
+}",
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a || b)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact]
+        public async Task MergedOnElseIfHeaderSelection3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(bool a, bool b)
+    {
+        if (a)
+        {
+        }
+[|        else if (b)
+|]        {
```

Comment:
I'm not sure I've seen this kind of selection before, but the one I do all the time is:

```
        }[|
        else if (b)|]
        {
```

(click somewhere past the end of the `else if` line, drag straight up a single line)

Reply:
This is a selection of the whole line (including the line break) that you'll get when you click in the left margin in the editor. What I like about this is that it matches up with how trivia around tokens work. Currently the logic is "is the selection around the header from `else` to `)`, but within the trivia", which makes the code simple and it also works pretty nicely. In this case it works because the text before the `else` falls within its trivia and the line break falls within trivia of the `)`.

I would not go out of my way to support odd selections like this unless other people have the same viewpoint. Maybe you can later send a PR to tweak this is the current supported selections are not good enough for you.

Reply:
Also, with the selection you showed, the cursor is on the previous line, not at the else if. The refactoring is always only offered on the second/inner if, and searches upward for a matching if statement.